### PR TITLE
ReactionStore migration to return plain errors

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `ReactionStore` to return plain errors

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/14818

### CODE REVIEW

**Comment by rvillablanca** [comment]
We should add the appropiate tags here :) @agnivade

File: app/export.go [hunk 1]
```diff
@@ -403,13 +403,14 @@ func (a *App) buildPostReplies(postId string) (*[]ReplyImportData, *model.AppErr
...
-	reactions, err := a.Srv().Store.Reaction().GetForPost(postId, true)
-	if err != nil {
-		return nil, err
+	reactions, nErr := a.Srv().Store.Reaction().GetForPost(postId, true)
+	if nErr != nil {
+		return nil, model.NewAppError("BuildPostReactions", "app.reaction.get_for_post.app_error", nil, nErr.Error(), http.StatusInternalServerError)
 	}
 
 	for _, reaction := range reactions {
 		var user *model.User
```
Comment (jespino) [inline app/export.go:412]: I think we can remove this 2 lines and use `:=` in the assignation below.

File: store/sqlstore/supplier_reactions.go [hunk 2]
```diff
@@ -119,19 +119,15 @@ func (s *SqlReactionStore) DeleteAllWithEmojiName(emojiName string) *model.AppEr
 				Reactions
 			WHERE
 				EmojiName = :EmojiName`, map[string]interface{}{"EmojiName": emojiName}); err != nil {
-		return model.NewAppError("SqlReactionStore.DeleteAllWithEmojiName",
-			"store.sql_reaction.delete_all_with_emoji_name.get_reactions.app_error", nil,
-			"emoji_name="+emojiName+", error="+err.Error(), http.StatusInternalServerError)
+		return errors.Wrapf(err, "failed to get Reacions with emijiName=%s", emojiName)
```
Comment (agnivade) [inline store/sqlstore/supplier_reactions.go:122]: typo at emoji, and also at Reactions

File: store/sqlstore/supplier_reactions.go [hunk 3]
```diff
@@ -119,19 +119,15 @@ func (s *SqlReactionStore) DeleteAllWithEmojiName(emojiName string) *model.AppEr
...
 
 	if _, err := s.GetMaster().Exec(
 		`DELETE FROM
 				Reactions
 			WHERE
 				EmojiName = :EmojiName`, map[string]interface{}{"EmojiName": emojiName}); err != nil {
-		return model.NewAppError("SqlReactionStore.DeleteAllWithEmojiName",
-			"store.sql_reaction.delete_all_with_emoji_name.delete_reactions.app_error", nil,
-			"emoji_name="+emojiName+", error="+err.Error(), http.StatusInternalServerError)
+		return errors.Wrapf(err, "failed to delete Reactions with emijiName=%s", emojiName)
```
Comment (agnivade) [inline store/sqlstore/supplier_reactions.go:130]: typo at emoji

File: app/reaction.go [hunk 4]
```diff
@@ -52,15 +59,19 @@ func (a *App) SaveReactionForPost(reaction *model.Reaction) (*model.Reaction, *m
 }
 
 func (a *App) GetReactionsForPost(postId string) ([]*model.Reaction, *model.AppError) {
-	return a.Srv().Store.Reaction().GetForPost(postId, true)
+	reaction, err := a.Srv().Store.Reaction().GetForPost(postId, true)
```
Comment (agnivade) [inline app/reaction.go:62]: reactions

File: app/reaction.go [hunk 5]
```diff
@@ -115,7 +126,7 @@ func (a *App) DeleteReactionForPost(reaction *model.Reaction) *model.AppError {
 	}
 
 	if _, err := a.Srv().Store.Reaction().Delete(reaction); err != nil {
-		return err
+		return model.NewAppError("deleteReactionForPost", "app.reaction.delete_all_with_emoji_name.get_reactions.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (agnivade) [inline app/reaction.go:129]: DeleteReactionForPost

**Comment by rvillablanca** [comment]
I had some ugly merge conflicts so let's wait how the tests end

**Comment by agnivade** [comment]
It was due to my changes. Sorry about that.

**Comment by rvillablanca** [comment]
> It was due to my changes. Sorry about that.

No worries, I love solving conflicts :joy:

File: store/sqlstore/supplier_reactions.go [hunk 6]
```diff
@@ -6,11 +6,11 @@ package sqlstore
 import (
 	"net/http"
 
+	"github.com/mattermost/gorp"
```
Comment (agnivade) [inline store/sqlstore/supplier_reactions.go:9]: This is an unnecessary change.

Comment (rvillablanca) [reply store/sqlstore/supplier_reactions.go]: Ok

**Comment by agnivade** [comment]
@lindy65 - All testing related to emoji reactions is needed here. Thanks.

**Comment by agnivade** [comment]
@rvillablanca - There was a minor issue in the merge conflict which I have fixed now. FYI.

**Comment by rvillablanca** [comment]
Thanks

**Comment by agnivade** [comment]
This needs some fixes in the EE repo. I'll make the changes Monday.

**Comment by rvillablanca** [comment]
Perfect thanks, but is any error related with this PR or it must fixed only in the EE repo? Let me know if I can help

**Comment by agnivade** [comment]
Only in EE repo. Basically some store calls in the EE repo need to be migrated to have the new errors.

**Comment by agnivade** [comment]
Closing this in favor of https://github.com/mattermost/mattermost-server/pull/14931.
